plutus-ir/src/Language/PlutusIR/Compiler/Datatype.hs

```diff
@@ -53,41 +55,96 @@ you can do with a datatype value: pattern match on it. A datatype value is there
 precisely the thing you need to do a pattern match. Namely, a function that takes functions implementing
 each arm of the match, and then gives you a result.
 
-For example, 'Just ' and 'Nothing' are encoded as
-\arg1 . \justBranch nothingBranch . justBranch arg1
-\justBranch nothingBranch . nothingBranch
-
 We also need to think about the types. In general, we need:
 - The type of the datatype
 - The type of each constructor
 - The value of each constructor
+- The type of the destructor
+- The value of the destructor
+
+-- Basic example
+
+For example, consider 'Maybe a'. Then:
+- The type of the datatype is:
+  Maybe = \(a :: *) -> forall out_Maybe . out_maybe -> (a -> out_maybe) -> out_maybe
```

Comment:
A typo: `out_Maybe` vs `out_maybe`. Do you mean this is how the data type looks like being Scott-encoded?

Reply:
Yes.